src/xinterpreter_raw.cpp

```diff
@@ -0,0 +1,348 @@
+
+#include "xeus-python/xinterpreter_raw.hpp"
+#include "xeus-python/xeus_python_config.hpp"
+#include "xeus-python/xtraceback.hpp"
+#include "xeus-python/xutils.hpp"
+
+#include "xcomm.hpp"
+#include "xkernel.hpp"
+#include "xdisplay.hpp"
+#include "xinput.hpp"
+#include "xinternal_utils.hpp"
+#include "xstream.hpp"
+#include "xinspect.hpp"
+
+namespace py = pybind11;
+namespace nl = nlohmann;
+using namespace pybind11::literals;
+
+namespace xpyt
+{
+
+    raw_interpreter::raw_interpreter(bool redirect_output_enabled /*=true*/, bool redirect_display_enabled /*=true*/)
+        
+    {
+        std::cout << "in raw mode" << std::endl;
+        xeus::register_interpreter(this);
+        if (redirect_output_enabled)
+        {
+            redirect_output();
+        }
+    }
+
+    raw_interpreter::~raw_interpreter()
+    {
+    }
+
+    void raw_interpreter::configure_impl()
+    {
+        if (m_release_gil_at_startup)
+        {
+            // The GIL is not held by default by the interpreter, so every time we need to execute Python code we
+            // will need to acquire the GIL
+            m_release_gil = gil_scoped_release_ptr(new py::gil_scoped_release());
+        }
+
+        py::gil_scoped_acquire acquire;
+
+        // py::module sys = py::module::import("sys");
+        // py::module logging = py::module::import("logging");
+
+        // py::module jedi = py::module::import("jedi");
+        // jedi.attr("api").attr("environment").attr("get_default_environment") = py::cpp_function([jedi]() {
+        //     jedi.attr("api").attr("environment").attr("SameEnvironment")();
+        // });
+
+        py::globals()["_i"] = "";
+        py::globals()["_ii"] = "";
+        py::globals()["_iii"] = "";
+
+    }
+
+    nl::json raw_interpreter::execute_request_impl(
+        int execution_count,
+        const std::string& code,
+        bool silent,
+        bool store_history,
+        nl::json user_expressions,
+        bool allow_stdin)
+    {
+        py::gil_scoped_acquire acquire;
+        nl::json kernel_res;
+        py::str code_copy;
+        // Scope guard performing the temporary monkey patching of input and
+        // getpass with a function sending input_request messages.
+        auto input_guard = input_redirection(allow_stdin);
+        if (code.size() >= 2 && code[0] == '?')
+        {
+            std::string result = formatted_docstring(code);
+            if (result.empty())
+            {
+                result = "Object " + code.substr(1) + " not found.";
+            }
+
+            kernel_res["status"] = "ok";
+            kernel_res["payload"] = nl::json::array();
+            kernel_res["payload"][0] = nl::json::object({
+                {"data", {
+                    {"text/plain", result}
+                }},
+                {"source", "page"},
+                {"start", 0}
+            });
+            kernel_res["user_expressions"] = nl::json::object();
+
+            return kernel_res;
+        }
```

Comment:
I would not keep this fake support, let's keep it simple and just do the execution